### DISCUSSION TITLE
Rj45 Breakout

### DISCUSSION THREAD

**Discussion by kreefd** [RChadwick7/FreedomWavePD#8] [Q&A]
Hi, 
I have a question about wiring/soldering the acuators with the rj45 breakouts. I have soldered one wire with - and one on + on the acuator. Now i have 8 + an 8 - cables how must i connect them to the breakout board is it relevant which (- or +) i connect to 1/2 , 3/4 , 4/5, 6/ ?

Greetings
Kreefd

**Comment by RChadwick7** [comment]
The polarity doesn't matter, but having them all the same probably is.  So, if the breakout is numbered 1-8, you could make 1,3,5,7 positive, and 2,4,6,8 negative, or 1,3,5,7 negative and 2,4,6,8 positive. Even this is not likely critical but that's how I wire mine.

**Comment by kreefd** [comment]
I tried both options but i am not able to run the actuators, on startup on the display appears Error 1.225-1 to 1.225-7 and then Error2-5 and i can see in very light text written DRV2605L but the text DRV2605L it disappears quickly  while Error2-5 is clearly to see.

Afterwards the Board Starts and Writes FreedomWavePD Ver.0.9.8

Might there be a problem with the drv2605L ?

I am using the LRA TEAX09C005 actuators.

**Comment by kreefd** [comment]
Update:

It seems to be the DRV2605L its not getting detected when i run a I2C Scan. Only on Adress 0x27 there is one detected i guess this is the LCD-Display ?

I tested with the multimeter the GND + SDA, GND + SCL, and GND + VIN they all work on all drivers, but GND + IN only works on 3 Drivers the rest is showing 0.0 on the multimeter.
I check all solderpoints and also resoldered them.

The strange thing was that GND + IN only worked on 2 drivers, I removed then one driver and then GND + IN suddenly worked on another one that it hadn't worked on before.

After I soldered the one I had taken out back on, 3 x drivers GND + IN suddenly worked

**Reply by RChadwick7** [reply]
I really should have documented those error codes better. Error 1.xxx means an issue with the  TCA9548A. The control signal first goes to the TCA9548A, then it goes to the DRV2605L. If there's a problem with the TCA9548A, it would make sense there would be issues communicating with the DRV2605L's. Is the orientation of the TCA9548A correct? Do you have a spare to replace it?

**Reply by RChadwick7** [reply]
I had a little more time to review what you wrote. Since you're talking about I2C, I'll assume you know what you're doing. :)

The LCD may very well be at 0x27

The DRV2605L's all have the same hardcoded address. The way I worked around this is to use the TCA9548A, which is used as an I2C multiplexer. If I want to talk to the first driver IC, I tell the multiplexer to switch to channel 1, then I talk to chip. I added a multiplexer test routine that writes to the multiplexer chip what channel it should talk to, then reads that value back. If the channel set doesn't equal the channel read, it throws an Error 1. If this test fails, it's very likely the later driver chip tests will fail. This seems to be where you are at.

I'd check pullup values. I mostly ignored I2C pullups, as the modules I've seen have them already. If you're not measuring 5V on the SCL and SDA, you might need to add pullup resistors. 

On the DRV2605L's, the IN pin is what actually triggers the driver output. If this is low, it just means the microcontroller on the Arduino didn't turn it on yet. It's odd you see a voltage on other pins but this could be timing. If treatment was active, all IN pins should go high at least briefly.

**Reply by kreefd** [reply]
To be honest i just asked ChatGPT how to check with an arduino the drivers but i dont really know much about Arduinos or I2C.

But i now unsoldered the controller you mentioned in the post before (it was a mess).
So i am now back home to solder a new one on the Board i hope it will work then.

What is a pull up resistor? Sorry but i have no or only little experience of electronics.

Thank you for taking time and responding.

**Reply by RChadwick7** [reply]
First, I'm happy to help in any way I can. I want you to be able to make these successfully. Beyond that, seeing your problems will help me make changes that will make it easier for others. Because of you, I've already added to my list of future changes and additions. Please ask for help if you need it.

The way I2C works is it communicates by only pulling down a pin that is naturally high. A resistor pulls this up to around 5V in this case. For I2C to work, having resistors on the SDA and SCL pins is critical, but most modules, like the LCD display, driver modules, and multiplexer module have these. The short version? If you measure 5 Volts on the SCL and SDA pins, you're probably good. If it's anywhere between 0 and 3V, there's likely a problem somewhere, and it might need two resistors to be added. It's not documented, but there is room on the PCB for them so they are easy to add. It could also be a lot of other things as well, like bad components, solder shorts, etc. Let me know what you find.

Also, to complicate things, while the Multimeter is an awesome tool, it won't be showing you the full picture. A pin might be swinging fast from 0 to 5V, and this might be normal. Your multimeter might read this as 2.5V, which would incorrectly indicate a problem. I'd recommend waiting 10 seconds after powering up to measure anything. Assuming treatment isn't running, all the SCL and SDA lines should have little if any data going on and you'll be more likely to get an accurate reading. When you first power up, it is checking and configuring a lot of things and your measurements might be deceptive at that time.

**Comment by kreefd** [comment]
So i now the I2C scan at least detects a second thing.

I2C Scanner
Scanning...
I2C device found at address 0x27  ! (before)
I2C device found at address 0x70  ! (new TCA9548A).


Before the replacement of the TCA9548A i had 5V on all SCL and SDA, now i have on almost all excpet U5 and U7.

U5: SCL = 3,70
      SDA = 4,95
      
U7 : SCL = 2,80
       SDA = 2,80
       
       
On the display it's writing Testing DRV2605L and the "L" LED on the Arduino is blinking fast but nothing else happens the menu doesn't appear neither the beep sounds.


Where can i put the mentioned resitors ? Is it the same ones like the Brown Red Brown on R13

**Reply by RChadwick7** [reply]
The pullup resistors go at R11 and R12, but that doesn't look like it will help. Those will only work for everything up to the TCA9548A. I didn't make room for pullup resistors between the TCA9548A and the DRv2605Ls. However, those resistors should be on the driver modules. Do you have any outputs working? Pictures might help.

**Comment by kreefd** [comment]
Oh okay. Do you mean by "Do you have any outputs working?" The GND + IN measurement ? Yes its working but only on U1, U2, and U5

On this picture you can see how SCL is lower and but on the same driver SDA is showing 4,95V
![20240805_114047](https://github.com/user-attachments/assets/e6ef621b-05ef-4168-abbb-98d78b9fbe5f)

Then the driver U7 SCL,

![20240805_114100](https://github.com/user-attachments/assets/a992598a-9038-4852-8fd6-3ed8942244be)

and SDA showing
![20240805_114105](https://github.com/user-attachments/assets/49109999-e4d5-4e54-82bf-85e66148285b)

The other drivers show for SDA an SCL 4,95 - 4.98 V.

So the board is staying like that with the Text "Testing DRV2605L"
![20240805_112859](https://github.com/user-attachments/assets/c4e8f2f8-0976-43fb-9e68-835f1add79ca)


I forgot to mention that i had by mistake ordered capacitors with 1500µF but you said in the video it wouldn't be a problem when i understood right.

**Comment by kreefd** [comment]
I made some more picture but i will put them here in this post. I checked it myself but i cant see anything wrong or bridges.

Top:
![20240805_111900](https://github.com/user-attachments/assets/cc69497a-7a28-4729-bac0-c03d085a4ea1)



![20240805_111951](https://github.com/user-attachments/assets/7a0fe197-528b-48ef-980d-edc5a0af89f1)


![20240805_112010](https://github.com/user-attachments/assets/dfbcfc5f-7911-4ff8-8727-c7e246fbe427)


![20240805_112037](https://github.com/user-attachments/assets/2e00e410-a452-41b2-9566-c375c4c78b99)


![20240805_112048](https://github.com/user-attachments/assets/14db0905-2ccb-4c29-84a6-ea9fe501156e)

![20240805_112055](https://github.com/user-attachments/assets/d638c210-cabc-4a11-b526-ff9e9617f655)


![20240805_112109](https://github.com/user-attachments/assets/0047c2d7-e023-45f0-8887-3866672f3575)

![20240805_112126](https://github.com/user-attachments/assets/295feff3-d402-4297-b9e4-7dbb17e60d6e)

**Comment by RChadwick7** [comment]
Thanks for the pictures, that helps. The first thing I would do is resolder everything. While some solder joints look good, others look like they could cause problems. If you have or can get flux paste, using that would be recommended. This is what I'd do:
1) Put solder flux paste on all solder joints
2) Set the soldering iron to 350c
3) Touch the soldering iron (no extra solder) to each joint for not more than 4 seconds. If you see the solder melt, wait 1-2 seconds then go to the next one.
4) Using a horsehair brush (Sometimes called acid brushes. They are cheap) and alcohol (The more pure the better. 91% is fairly easy to find. 70% won't work as fast), and scrub all solder joints, especially between pins. When done, do again with a clean brush and more alcohol to get all the flux off.

When done, the board should be clean and every solder joint should be shiny and all should pretty much look the same.

Soldering issues can create all kinds of problems that are strange and difficult to diagnose. I have seen fine particles of solder trapped in flux that will partially conduct, even if you don't see a solder bridge with your eye. It is also easy to partially solder a connection where things sometimes work, or partially works.

If you have any questions about this, please ask.

**Comment by RChadwick7** [comment]
Also, I see what looks like a solder bridge on the U7 module. This can definitely cause at least part of your issue. I'd recommend looking carefully, with some kind of magnification, across the entire board, including the modules. With flux and good technique, the solder bridges on IC's with multiple pins can be repaired easily. Basically just put flux on the bridge, heat up your soldering iron, wipe off any solder on the iron's tip, and swipe the solder bridge away, swiping the soldering iron tip with the direction of the pins. The swipe should take about 2-3 seconds. If the first time doesn't work, you can try swiping again. Don't leave the tip longer than 5 seconds or you may damage the PCB or completely unsolder the part. Be VERY careful of smaller components. With heat, these come up very easily but it takes a bit of skill to get them back. Use this technique only on parts with multiple pins, like the black IC on U7. It might be safest to take pictures of any solder bridges you find and we can come up with a safe gameplan on how to fix them.

**Reply by kreefd** [reply]
Oh yes i can see it now thank you. you are right. This little drop of solder has to be got drop there while soldering.
Here is picture with zoom.

![image](https://github.com/user-attachments/assets/8396bd75-4a97-47e6-b3cf-d0a3325bcb88)


But its very hard to remove it i am afraid of losen up the little chip.

Unfortunatelx i have no flux at the moment. I need

**Reply by kreefd** [reply]
Hello,
 
I ordered new drivers and finally got it to work.
Thank you very much for your assitance i will report for documentation in 4,8 and 12 weeks which improvements my father may have.

God or any entity you believe in, bless you. :)

**Reply by RChadwick7** [reply]
I'm glad you got it together. I hope it helps!
